packages/cli/src/metrics/license-metrics.service.ts

```diff
@@ -37,7 +37,8 @@ export class LicenseMetricsService {
 
 	async collectPassthroughData() {
 		return {
-			activeWorkflowIds: await this.workflowRepository.getActiveIds(),
+			// Get only the first 1000 active workflow IDs to avoid sending too much data to License Server
+			activeWorkflowIds: (await this.workflowRepository.getActiveIds()).slice(0, 1000),
```

Comment:
we should fetch only 1000 ids in the query here, instead of truncating the array in the application.